authorization-service/src/functions/basicAuthorizer/handler.ts

```diff
@@ -0,0 +1,69 @@
+import 'source-map-support/register';
+import {
+  APIGatewayTokenAuthorizerHandler,
+  APIGatewayTokenAuthorizerEvent,
+  APIGatewayAuthorizerResult,
+  Callback
+} from 'aws-lambda';
+
+import { middyfy } from '@libs/lambda';
+
+const basicAuthorizer: APIGatewayTokenAuthorizerHandler = async (
+  event: APIGatewayTokenAuthorizerEvent, _, cb: Callback<APIGatewayAuthorizerResult>
+): Promise<any> => {
+  console.log('Lambda invokation "basicAuthorizer": ', event);
+
+  if (event.type !== 'TOKEN') {
+    cb('Unauthorized');
+  }
+
+  try {
+    const authorizationToken = event.authorizationToken;
+
+    const encodedCreds = authorizationToken.split(' ')[1];
+    const buff = Buffer.from(encodedCreds, 'base64');
+    const plainCreds = buff.toString('utf-8').split(':');
+
+    const username = plainCreds[0];
+    const password = plainCreds[1];
+
+    console.log(`username=${username} and password=${password}`);
+    const storedUserPassword = process.env[username];
+
+    if (!storedUserPassword) {
+      cb('Access Denied');
```

Comment:
For any token value other than "unauthorized", the authorizer returns an HTTP 500 status code. https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-use-lambda-authorizer.html

Reply:
ok, thanks